cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -125,14 +126,15 @@ func startNodeIpamController(initContext app.ControllerInitContext, ccmConfig *c
 
 	var clusterCIDRInformer v1alpha1.ClusterCIDRInformer
 	if utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRRangeAllocator) {
-		clusterCIDRInformer = ctx.InformerFactory.Networking().V1alpha1().ClusterCIDRs()
+		clusterCIDRInformer = controllerCtx.InformerFactory.Networking().V1alpha1().ClusterCIDRs()
 	}
-
+	ctx, _ := wait.ContextForChannel(wait.NeverStop)
```

Comment:
Instead of creating a new context inside this function, let the caller do that and pass it in.

This then bubbles up all the way to `cmd/kube-controller-manager/app/core.go` where you can add a WithName for the controller as as in https://github.com/kubernetes/kubernetes/pull/113622/files#r1026670688.

If you need a context in a test, use ktesting. See the PR above for an example.